sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_file_client.py

```diff
@@ -597,6 +597,49 @@ def download_file(self, offset=None, length=None, **kwargs):
         downloader = self._blob_client.download_blob(offset=offset, length=length, **kwargs)
         return StorageStreamDownloader(downloader)
 
+    def create_snapshot(self, metadata=None, **kwargs):
```

Comment:
do you happen to know if service supports snapshot directory? like snapshot the whole directory including the blob.

Reply:
I have just attempted it - it's not possible.